.ci/Jenkinsfile

```
@@ -287,3 +302,91 @@ def archiveArtifactsSafe(remotePath, artifacts) {
     credentialsId: "${JOB_GCS_EXT_CREDENTIALS}",
     pattern: artifacts)
 }
+
+def packageStoragePublish() {
+  stage('Publish to Package Storage v2') {
+    if (env.BRANCH_NAME != 'main') {
+      echo "packageStoragePublish: not the main branch, nothing will be published"
+      return
+    }
+
+    deleteDir()
+    unstashV2(name: 'source', bucket: "${JOB_GCS_BUCKET}", credentialsId: "${JOB_GCS_CREDENTIALS}")
+    useElasticPackage()
+
+    // Build only unpublished files (need to read version from manifest.yml)
+    dir("${BASE_DIR}/packages") {
+      findFiles()?.findAll{ !it.name.endsWith('@tmp') }?.collect{ it.name }?.sort()?.each {
+        dir("${it}") {
+          def manifest = readYaml(file: "manifest.yml")
+          if (manifest != null && manifest["version"] != null) {
+            def packageZip = it + "-" + manifest["version"] + ".zip"
+            if (isAlreadyPublished(packageZip)) {
+              return
+            }
+          }
+
+          sh(label: "Build integration as zip: ${it}", script: """
+            PATH=${env.WORKSPACE}/${BASE_DIR}/build:${env.PATH}
+            elastic-package build --zip""")
```

Nit. Could `withEnv` be used to set this variable?
```suggestion
          withEnv(["PATH=${env.WORKSPACE}/${BASE_DIR}/build:${env.PATH}"]) {
            sh(label: "Build integration as zip: ${it}", script: "elastic-package build --zip")
          }
```

---

Fixed